UhooiPicBook/SceneDelegate.swift

```diff
@@ -55,3 +61,34 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     }
 
 }
+
+// MARK: - UserActivity
+extension SceneDelegate {
+
+    func scene(_ scene: UIScene, continue userActivity: NSUserActivity) {
+        self.executeUserActivity(userActivity)
+    }
+
+    private func executeUserActivity(_ userActivity: NSUserActivity) {
+        switch userActivity.activityType {
+        case CSSearchableItemActionType:
+            self.executeSpotlightActivity(userActivity)
+        default:
+            return
+        }
+    }
+
+    private func executeSpotlightActivity(_ userActivity: NSUserActivity) {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+
+        guard let userDefaultsKey = userActivity.userInfo?[CSSearchableItemActivityIdentifier] as? String,
+            let data = UserDefaults.standard.value(forKey: userDefaultsKey) as? Data,
```

Comment:
型が決まっている場合は予め用意されているメソッドを使うべきだと思います！

```suggestion
            let data = UserDefaults.standard.data(forKey: userDefaultsKey),
```